src/sentry/auth/helper.py

```diff
@@ -663,42 +668,61 @@ def get_for_request(cls, request):
             provider_key=req_state.provider_key,
         )
 
-    def __init__(self, request, organization, flow, auth_provider=None, provider_key=None):
+    def __init__(
+        self,
+        request: HttpRequest,
+        organization: Organization,
+        flow: int,
+        auth_provider: AuthProvider | None = None,
+        provider_key: str | None = None,
+    ) -> None:
         assert provider_key or auth_provider
         self.flow = flow
-        super().__init__(request, provider_key, organization, auth_provider)
 
-    def get_provider(self, provider_key):
+        # TODO: Resolve inconsistency with nullable provider_key.
+        # Tagging with "type: ignore" because the superclass requires provider_key to
+        # be non-nullable. We get away with it because super().__init__ only passes
+        # provider_key to get_provider, and our get_provider override accepts a null
+        # provider_key. But it technically violates the type contract and we'll need
+        # to change the superclass to accommodate this one.
+        super().__init__(request, provider_key, organization, auth_provider)  # type: ignore
+
+        # Override superclass's type hints to be narrower
+        self.organization: Organization = self.organization
```

Comment:
Yup, noticed this.  The organization is None for Pipelines at the start, but initializing state loads this in.  Kind of an odd flow.